lib/rabbitmq/cli/plugins/commands/directories_command.ex

```diff
@@ -0,0 +1,119 @@
+## The contents of this file are subject to the Mozilla Public License
+## Version 1.1 (the "License"); you may not use this file except in
+## compliance with the License. You may obtain a copy of the License
+## at http://www.mozilla.org/MPL/
+##
+## Software distributed under the License is distributed on an "AS IS"
+## basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See
+## the License for the specific language governing rights and
+## limitations under the License.
+##
+## The Original Code is RabbitMQ.
+##
+## The Initial Developer of the Original Code is Pivotal Software, Inc.
+## Copyright (c) 2007-2018 Pivotal Software, Inc.  All rights reserved.
+
+
+defmodule RabbitMQ.CLI.Plugins.Commands.DirectoriesCommand do
+  alias RabbitMQ.CLI.Plugins.Helpers, as: PluginHelpers
+  alias RabbitMQ.CLI.Core.{Helpers, Validators}
+
+  @behaviour RabbitMQ.CLI.CommandBehaviour
+
+  def formatter(), do: RabbitMQ.CLI.Formatters.String
+
+  def merge_defaults(args, opts) do
+    {args, Map.merge(%{online: true, offline: false}, opts)}
+  end
+
+  def distribution(%{offline: true}),  do: :none
+  def distribution(%{offline: false}), do: :cli
+
+  def switches(), do: [online: :boolean,
+                       offline: :boolean]
+
+  def validate(_, %{online: true, offline: true}) do
+   {:validation_failure, {:bad_argument, "Cannot set both online and offline"}}
+  end
+  def validate(_, %{online: false, offline: false}) do
+   {:validation_failure, {:bad_argument, "Cannot set online and offline to false"}}
+  end
+  def validate([_ | _], _) do
+    {:validation_failure, :too_many_args}
+  end
+  def validate([], _) do
+    :ok
+  end
+
+  def validate_execution_environment(args, %{offline: true} = opts) do
+    Validators.chain([&Helpers.require_rabbit_and_plugins/2,
+                      &PluginHelpers.enabled_plugins_file/2,
+                      &Helpers.plugins_dir/2],
+                     [args, opts])
+  end
+  def validate_execution_environment(args, %{offline: false} = opts) do
+    Validators.node_is_running(args, opts)
+  end
+  def validate_execution_environment(args, %{online: true} = opts) do
+    Validators.node_is_running(args, opts)
+  end
+
+  def usage, do: "directories [--offline] [--online]"
+
+  def banner([], %{online: false, offline: true}) do
+    "Listing plugin directories..."
+  end
+
+  def banner([], %{online: true, offline: false, node: node}) do
+    "Listing plugin directories used by node #{node}"
+  end
+
+  def run([], %{online: true, node: node_name}) do
+    do_run fn(key) ->
+      :rabbit_misc.rpc_call(node_name, :rabbit_plugins, key, [])
+    end
+  end
+
+  def run([], %{offline: false, node: node_name}) do
```

Comment:
Same as the previous comment. This clause should be matched in the previous one.